### PR TITLE
[studio-ui] Updating LegacyFormDialog;

### PR DESCRIPTION
**Pull request by aachavarria**
Updating LegacyFormDialog.tsx to show it on a Dialog;
Adding success messages on wcm_content.js and wcm-dashboardwidget-common.js Edit;

### CODE REVIEW

File: ui/app/src/state/epics/system.ts [hunk 1]
```diff
@@ -83,11 +83,17 @@ const systemEpics: CrafterCMSEpic[] = [
   (action$, state$, { getIntl }) =>
     action$.pipe(
       ofType(showEditItemSuccessNotification.type),
-      tap(({ payload }) => {
+      tap(({ payload: { action } }) => {
+        let message;
+        if (['save', 'saveAndMinimize'].includes(action)) {
+          message = getIntl().formatMessage(itemSuccessMessages.itemSavedAsDraft);
```
Comment (rart) [inline ui/app/src/state/epics/system.ts]: These two actions do regular save. Why showing save draft message?

File: ui/app/src/utils/i18n-legacy.ts [hunk 2]
```diff
@@ -1439,6 +1439,10 @@ export const itemSuccessMessages = defineMessages({
   itemContentTypeChanged: {
     id: 'item.contentTypeChanged',
     defaultMessage: 'Item type changed successfully'
+  },
+  itemSavedAsDraft: {
+    id: 'item.savedAsDraft',
+    defaultMessage: 'Item saved as draft'
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: Item draft saved

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/epics/system.ts (hunk 1 above)
Comment (aachavarria) [reply ui/app/src/state/epics/system.ts]: fixed

File: ui/app/src/utils/i18n-legacy.ts [hunk 3]
```diff
@@ -690,17 +686,17 @@ export const browseSearchMessages = defineMessages({
 });
 
 export const formEngineMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'formEngine.inProgressConfirmation',
+    defaultMessage: 'There are an edit in progress, to open a new edit form please close or save the previous one'
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: Message: "Another form is currently open & minimized. Please close the current form before opening another."

We need this case for template editor too.

File: ui/app/src/utils/codebase-bridge.tsx [hunk 4]
```diff
@@ -245,7 +245,11 @@ export function createCodebaseBridge() {
           // @ts-ignore
           ReactDOM.render(
             // @ts-ignore
-            <CrafterCMSNextBridge mountGlobalDialogManager={!isLegacy} mountSnackbarProvider={!isLegacy}>
+            <CrafterCMSNextBridge
+              mountGlobalDialogManager={!isLegacy}
+              mountSnackbarProvider={!isLegacy}
+              mountLegacyConcierge={isLegacy}
+            >
```
Comment (rart) [inline ui/app/src/utils/codebase-bridge.tsx]: PR mixup 😣

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 5]
```diff
@@ -771,7 +772,7 @@ var CStudioForms =
     }
 
     function saveDraftDialog() {
-      $.notify(formatMessage(formEngineMessages.saveDraftCompleted), 'success');
+      // deprecated
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js]: ?

File: ui/app/src/utils/codebase-bridge.tsx (hunk 4 above)
Comment (aachavarria) [reply ui/app/src/utils/codebase-bridge.tsx]: no idea

Comment (aachavarria) [reply ui/app/src/utils/codebase-bridge.tsx]: this is already on develop, should not be listed here

File: ui/app/src/utils/i18n-legacy.ts [hunk 6]
```diff
@@ -534,6 +534,10 @@ export const siteComponentDSMessages = defineMessages({
 });
 
 export const codeEditorMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'codeEditor.inProgressConfirmation',
+    defaultMessage: 'Another form is currently open & minimized. Please close the current form before opening another'
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: "Another **editor** is currently..."

File: ui/app/src/state/epics/dialogs.ts [hunk 7]
```diff
@@ -125,6 +130,58 @@ const dialogEpics: CrafterCMSEpic[] = [
...
+              })
+            );
+          } else {
+            return NEVER;
+          }
+        } else {
+          return of(
+            showConfirmDialog({
+              body: getIntl().formatMessage(formEngineMessages.inProgressConfirmation)
+            })
```
Comment (rart) [inline ui/app/src/state/epics/dialogs.ts]: The confirm dialog presents a question and this isn't one. We need to change the graphic and only the accept button should show up. See Confirm Dialog designs on Figma (3.2 project) for guidance.

We need to tweak ConfirmDialog to allow graphic swapping.
```jsx
function ConfirmDialogWrapper(props: ConfirmDialogProps) {
  const { onOk, onCancel, body, title, children, classes, imageUrl = questionGraphicUrl } = props;
  const { formatMessage } = useIntl();
  useUnmount(props.onClosed);
  return (
    <>
      <DialogContent id="confirmDialogBody" className={classes.dialogBody}>
        <img src={imageUrl} alt="" />
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/utils/i18n-legacy.ts [hunk 8]
```diff
@@ -534,6 +534,10 @@ export const siteComponentDSMessages = defineMessages({
 });
 
 export const codeEditorMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'codeEditor.inProgressConfirmation',
+    defaultMessage: 'Another editor is currently open & minimized. Please close the current form before opening another'
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: "Please close the current **editor** before opening another"

File: ui/app/src/state/epics/dialogs.ts [hunk 9]
```diff
@@ -125,6 +131,59 @@ const dialogEpics: CrafterCMSEpic[] = [
...
+                minimized: false
+              })
+            );
+          } else {
+            return NEVER;
+          }
+        } else {
+          return of(
+            showConfirmDialog({
+              body: getIntl().formatMessage(codeEditorMessages.inProgressConfirmation)
```
Comment (rart) [inline ui/app/src/state/epics/dialogs.ts]: info graphic

File: ui/app/src/utils/i18n-legacy.ts [hunk 10]
```diff
@@ -690,17 +691,18 @@ export const browseSearchMessages = defineMessages({
 });
 
 export const formEngineMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'formEngine.inProgressConfirmation',
+    defaultMessage:
+      'Another editor is currently open & minimized. Please close the current editor before opening another'
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: This is a form. The other is an editor. This was was fine saying form in both sentences. The one above should say editor in both sentences.

File: ui/app/src/state/epics/dialogs.ts (hunk 9 above)
Comment (rart) [reply ui/app/src/state/epics/dialogs.ts]: Why is this resolved? It isn’t fixed yet.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@aachavarria please fix conflicts

**Comment by aachavarria** [comment]
> @aachavarria please fix conflicts

done

**Review by rart — APPROVED** [review]

**Pull request merged**